packages/roosterjs-editor-api/lib/utils/splitList.ts

```diff
@@ -0,0 +1,22 @@
+import blockFormat from '../utils/blockFormat';
+import { createVListFromRegion } from 'roosterjs-editor-dom';
+import { IEditor } from 'roosterjs-editor-types';
+
+/**
+ * @internal
+ */
+export default function splitList(
+    editor: IEditor,
+    list: HTMLOListElement,
+    startNumber?: number
+): void;
+
+export default function splitList(editor: IEditor, list: HTMLOListElement, startNumber?: number) {
+    blockFormat(editor, (region, start) => {
```

Comment:
blockFormat relies on current selection in editor, and it will handle complex selection case such as select across a table. Here we already know which list to operate on, so no need to call blockFormat, and we should use the list or li parameter directly.

Reply:
And actually I think we need to pass in the LI node instead list, then we can restart the list number from the passed in li node, which means if we have a list "1 2 3 4", and I point to "3" and reset the number, it should become "1 2 1 2".